packages/lib-classifier/src/store/Step.js

```diff
@@ -7,28 +7,50 @@ function taskDispatcher (snapshot) {
 
 const GenericTask = types.union({ dispatcher: taskDispatcher }, ...taskModels)
 
-export const BaseStep = types
-  .model('BaseStep', {
+const Step = types
+  .model('Step', {
+    next: types.maybe(types.string),
+    previous: types.maybe(types.string),
     stepKey: types.identifier,
     taskKeys: types.array(types.string),
     tasks: types.array(GenericTask)
   })
   .views(self => ({
     get isComplete () {
       return self.tasks.reduce((isStepComplete, task) => isStepComplete && task.isComplete, true)
+    },
+
+    get isThereANextStep () {
+      return !!self.next && self.next !== 'summary'
+    },
+
+    get isThereAPreviousStep () {
+      return !!self.previous && self.stepKey !== 'summary'
+    },
+
+    get isThereBranching () {
+      // We return the first single choice task
+      // It doesn't make sense to have more than one single choice task in a step
+      // so we should disallow that in the editor
+      const [singleChoiceTask] = self.tasks.filter(task => task.type === 'single')
+      return singleChoiceTask?.answers.some((answer, index) => {
+        if (singleChoiceTask.answers.length > index + 1) {
+          return answer.next !== singleChoiceTask.answers[index + 1].next
+        }
+        return false
+      })
     }
   }))
   .actions(self => ({
     reset () {
       self.tasks.forEach(task => task.reset())
```

Comment:
The step reset here also needs to reset the next and previous value for a step with branching logic. 
```suggestion
      self.tasks.forEach(task => task.reset())
      if (self.isThereBranching) {
        self.setNext(undefined)
        self.setPrevious(undefined)
      }
```

I found this issue when testing with Kitteh Zoo. The state of the previous annotation selection for which step is next remained with subsequent subjects, thus causing the 'next' button to display at the start instead of 'done', which is the current behavior. 

Apparently the spec file is missing a test for the reset action, so this needs to be added.